### PR TITLE
Update alz_checklist.en.json

### PR DESCRIPTION
**Pull request by mbilalamjad**
Adding a check to help Architects/Engineers have an overall conversation on the encryption on Azure with customers. This is based on feedback we received.

### CODE REVIEW

File: checklists/alz_checklist.en.json [hunk 1]
```diff
@@ -2253,14 +2253,26 @@
             "severity": "Medium",
             "link": "https://www.microsoft.com/security/business/zero-trust"
         },
+        {
+            "category": "Security",
+            "subcategory": "Encryption and keys",
+            "text": "Consider planning for encryption at rest and in transit for all data along with key management in Azure.",
```
Comment (brsteph) [inline checklists/alz_checklist.en.json:2259]: Can we change this into something that is more of a review item?

Something like "Configure encryption at rest and in transit on Key Vaults" and then linking to that specific setting.  A really broad "consideration" item probably isn't the best kind of item for the check list.  We are trying to capture guide lines for this here: https://github.com/Azure/review-checklists/blob/main/CONTRIBUTING.md#what-to-contribute

Comment (igorjnzl) [reply checklists/alz_checklist.en.json:2259]: I think the scope of services would be quite limited for recommendation purposes, we could add 1 or many specific ones. E.g;
Use Azure disk encryption or encryption at host for your IaaS workloads

Thoughts?

Comment (igorjnzl) [reply checklists/alz_checklist.en.json:2259]: I see your point, the feedback we got was that there are no checks/guidance around encryption, and it was highlighted as a gap.

Might be worth having a meeting and taking this offline for a discussion?

Comment (brsteph) [reply checklists/alz_checklist.en.json:2259]: Workshop text:

Deploy policies to workload subscription management groups to manage encryption at rest on resources that support it, such as managed disk encryption and enabling TDE on SQL instances.

Comment (erjosito) [reply checklists/alz_checklist.en.json:2259]: Where are we with this one?

**Comment by mbilalamjad** [comment]
@igorjnzl I'll close this PR due to change in priorities, feel free to to do a separate PR based on our last discussion on this.

**Pull request closed without merging**